### ISSUE TITLE
SmtpClient: getting 503 5.5.2 'Send hello first' when it's connecting to an Exchange server

### ISSUE THREAD

**Issue by pvanek** [qorelanguage/qore#2833]
Example:
```
SmtpClient s("esmtp://xxx:xxx@server", \logx(), \logx());
s.connect();

$ qore t.q
list: ("SmtpClient: SmtpClient server: \"server:587\"")
list: ("SmtpClient: set authentication credentials to xxx/xxx")
list: ("SmtpClient: connect reply: \"220 server Microsoft ESMTP MAIL Service ready at Tue, 22 May 2018 14:54:29 +0200\"")
list: ("SmtpClient: send \"EHLO  myhost\"")
list: ("SmtpClient: reply: \"250-server Hello [10.10.11.226]\"")
list: ("SmtpClient: reply: \"250-SIZE 52428800\"")
list: ("SmtpClient: reply: \"250-PIPELINING\"")
list: ("SmtpClient: reply: \"250-DSN\"")
list: ("SmtpClient: reply: \"250-ENHANCEDSTATUSCODES\"")
list: ("SmtpClient: reply: \"250-STARTTLS\"")
list: ("SmtpClient: reply: \"250-AUTH GSSAPI NTLM\"")
list: ("SmtpClient: reply: \"250-8BITMIME\"")
list: ("SmtpClient: reply: \"250-BINARYMIME\"")
list: ("SmtpClient: reply: \"250 CHUNKING\"")
list: ("SmtpClient: server supports STARTTLS; automatically setting STARTTLS flag")
list: ("SmtpClient: send \"STARTTLS\"")
list: ("SmtpClient: reply: \"220 2.0.0 SMTP server ready\"")
list: ("SmtpClient: upgrade to TLS/SSL connection ECDHE-RSA-AES128-SHA256 TLSv1/SSLv3")
list: ("SmtpClient: send \"AUTH PLAIN ZXNob3BAZHBiLnNrAGVzaG9wQGRwYi5zawBDMXA1aG9A\"")
list: ("SmtpClient: reply: \"503 5.5.2 Send hello first\"")
```

sending EHLO after rh = sendCommand("STARTTLS"); fixes it

```
                # issue STARTTLS and upgrade to a TLS connection if the tls flag is set and we are not already secure
                if (tls && !sock.isSecure()) {
                    rh = sendCommand("STARTTLS");
                    if (rh.code >= 500)
                        throw "SMTP-DATA-EXCEPTION", sprintf("error sending STARTTLS command: %y", rh);

                    sock.upgradeClientToSSL();
                    logDbg("upgrade to TLS/SSL connection %s %s", sock.getSSLCipherName(), sock.getSSLCipherVersion());

string hello = esmtp ? "EHLO" : "HELO";
hash rh = sendCommand(hello + "  %s", gethostname());
if (rh.code >= 500)
    throw "SMTP-DATA-EXCEPTION", sprintf("error sending %s command: %y", hello, rh);
}
```

refs eg. https://github.com/bertjohnson/OpaqueMail/issues/35

**Comment by davidnich** [comment]
@pvanek 0.8.13.5 has been released, we have to do this for 0.8.13.6; I'll close the 0.8.13.5 milestone now

**Comment by davidnich** [comment]
this bug will not happen when using the `smtps` URL scheme
